### PR TITLE
Accept comments

### PR DESCRIPTION
**Pull request by davidge20**
## Description

Fixes # (issue)

All functions related to and including accept_syscall were commented out

 

### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

Will add test cases if necessary

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been added to a pull request and/or merged in other modules (native-client, lind-glibc, lind-project)

### CODE REVIEW

**Comment by davidge20** [comment]
@ve1nard requesting your review, thanks

File: src/safeposix/syscalls/net_calls.rs [hunk 1]
```diff
@@ -1765,22 +1765,102 @@ impl Cage {
...
+    /// * EPERM - Firewall rules forbid connection.
+    /// 
+    /// * EPROTO - Protocol error.
+    /// 
+    /// In addition, network errors for the new socket and as defined for
+    /// the protocol may be returned.  Various Linux kernels can return
+    /// other errors such as ENOSR, ESOCKTNOSUPPORT, EPROTONOSUPPORT,
+    /// ETIMEDOUT.  The value ERESTARTSYS may be seen during a trace.
+    /// 
+    /// ### Panics
```
Comment (davidge20) [inline src/safeposix/syscalls/net_calls.rs:2271]: update panics

**Comment by rennergade** [comment]
@pranav-bhatt can you take a look at this one as well

File: src/safeposix/syscalls/net_calls.rs [hunk 2]
```diff
@@ -1765,22 +1765,102 @@ impl Cage {
...
 
+    /// ### Description
+    /// 
+    /// `accept_syscall` accepts a connection on a socket 
+    /// 
+    /// ### Arguments
+    /// 
+    /// it accepts two parameters: 
+    /// * `fd` - the file descriptor that refers to the listening socket
+    /// * `addr` - the address of the peer socket (i.e. the client that is connecting)
```
Comment (pranav-bhatt) [inline src/safeposix/syscalls/net_calls.rs]: might be pedantic, but do you think using connection directionality would be clearer? For example "the address of the peer socket (i.e. the address of the incoming connection)"

File: src/safeposix/syscalls/net_calls.rs [hunk 3]
```diff
@@ -1765,22 +1765,102 @@ impl Cage {
...
+    ///            // ** Do we deal with the case in which addr may be NULL ?? ** //
+    /// 
+    /// ### Returns
+    /// 
+    /// for a successful call, the return value will be a file descriptor for the
+    /// accepted socket (a nonnegative integer). On error, -errno is
```
Comment (pranav-bhatt) [inline src/safeposix/syscalls/net_calls.rs]: better phrasing for the error part? "On error, a negative error number is returned, with the `errorno` set to represent the corresponding error"

File: src/safeposix/syscalls/net_calls.rs [hunk 4]
```diff
@@ -1765,22 +1765,102 @@ impl Cage {
...
+    /// 
+    /// * ENOTSOCK - The file descriptor sockfd does not refer to a socket.
+    /// 
+    /// * EOPNOTSUPP - The referenced socket is not of type SOCK_STREAM.
+    /// 
+    /// * EPERM - Firewall rules forbid connection.
+    /// 
+    /// * EPROTO - Protocol error.
+    /// 
+    /// In addition, network errors for the new socket and as defined for
```
Comment (pranav-bhatt) [inline src/safeposix/syscalls/net_calls.rs:2264]: Should these also be documented in the above format if they are possible values that can be observed?

File: src/safeposix/syscalls/net_calls.rs [hunk 5]
```diff
@@ -1959,21 +2105,32 @@ impl Cage {
...
-                    let porttuple =
-                        mux_port(ladr.addr().clone(), ladr.port(), sockhandle.domain, TCPPORT);
 
-                    let mut pendingvec =
-                        NET_METADATA.pending_conn_table.get_mut(&porttuple).unwrap();
+                    //Socket must have been populated by implicit bind
+                    let ladr = sockhandle.localaddr.unwrap().clone();
+                    //Obtain a tuple of the address, port, and port type of the listening socket
+                    //Panics if domain is not INET or INET6
+                    let porttuple = mux_port(ladr.addr().clone(),
```
Comment (pranav-bhatt) [inline src/safeposix/syscalls/net_calls.rs]: has cargo fmt been run? this seems like odd formatting

File: src/safeposix/syscalls/net_calls.rs [hunk 6]
```diff
@@ -1765,22 +1765,102 @@ impl Cage {
...
+    /// ### Description
+    /// 
+    /// `accept_syscall` accepts a connection on a socket 
+    /// 
+    /// ### Arguments
+    /// 
+    /// it accepts two parameters: 
+    /// * `fd` - the file descriptor that refers to the listening socket
+    /// * `addr` - the address of the peer socket (i.e. the client that is connecting)
+    ///            // ** Do we deal with the case in which addr may be NULL ?? ** //
```
Comment (ve1nard) [inline src/safeposix/syscalls/net_calls.rs]: from what Nick explained to me, null pointers and other stuff is sorted out before it gets to the call, so we don't have to worry about those

File: src/safeposix/syscalls/net_calls.rs (hunk 4 above)
Comment (ve1nard) [reply src/safeposix/syscalls/net_calls.rs:2264]: the errors above are those that we support, the rest of the errors described below exist in the standard but are not supported within this syscall

File: src/safeposix/syscalls/net_calls.rs [hunk 7]
```diff
@@ -1822,6 +1902,17 @@ impl Cage {
         }
```
Comment (ve1nard) [inline src/safeposix/syscalls/net_calls.rs:2345]: should the syscall name in the parentheses be changed to "accept" instead of "listen"?

File: src/safeposix/syscalls/net_calls.rs [hunk 8]
```diff
@@ -1855,25 +1954,41 @@ impl Cage {
...
                 );
 
+                //Initialize pipes
+                //In the Unix domain, lind emulates communcation
                 let remote_addr: interface::GenSockaddr;
                 let sendpipenumber;
                 let receivepipenumber;
 
+                // We loop here to accept the connection.
+                //
```
Comment (ve1nard) [inline src/safeposix/syscalls/net_calls.rs]: i think the newlines here are unnecessary

File: src/safeposix/syscalls/net_calls.rs [hunk 9]
```diff
@@ -1893,39 +2009,62 @@ impl Cage {
...
+                    //Insert necessary info about the socket communication
                     newsockhandle.unix_info = Some(UnixSocketInfo {
                         inode: inodenum.clone(),
                         mode: sockhandle.unix_info.as_ref().unwrap().mode,
                         sendpipe: Some(sendpipenumber.clone()),
                         receivepipe: Some(receivepipenumber.clone()),
                     });
+                    //Grab the peer socket inode from the inodetable 
+                    //and increase the refcount by 1
+                    // ** Why is this necessary ?? ** //
```
Comment (ve1nard) [inline src/safeposix/syscalls/net_calls.rs]: not sure, but from what i saw while working with filesystem calls is that close_syscall() looks at the refcount of the file and proceeds with the removal if it is zero. i guess we don't want the open socket to be removed, that is why we increase its refcount.

File: src/safeposix/syscalls/net_calls.rs [hunk 10]
```diff
@@ -1765,22 +1765,102 @@ impl Cage {
         return 0;
     }
 
+    /// ### Description
```
Comment (ve1nard) [inline src/safeposix/syscalls/net_calls.rs:2207]: great job on providing detailed comments! what about the tests? do we have tests for these syscalls?

File: src/safeposix/syscalls/net_calls.rs (hunk 7 above)
Comment (davidge20) [reply src/safeposix/syscalls/net_calls.rs:2345]: good catch!

File: src/safeposix/syscalls/net_calls.rs [hunk 11]
```diff
@@ -1765,22 +1765,103 @@ impl Cage {
...
 
+            //Pattern match such that FileDescriptor object must be the Socket variant
+            //Otherwise, return with an err as the fd refers to something other than a socket
             match filedesc_enum {
                 Socket(ref mut sockfdobj) => {
+                    //Clone the socket handle as it may be in use by other threads and
+                    //obtain a read lock, blocking the calling thread until
+                    //there are no other writers that hold the lock
                     let sock_tmp = sockfdobj.handle.clone();
                     let mut sockhandle = sock_tmp.read();
```
Comment (davidge20) [inline src/safeposix/syscalls/net_calls.rs:2304]: Does it have to be a mutable read guard? I deleted the mut keyword and all the tests still passed. 

(Thank you Qianxi for pointing it out)

**Comment by davidge20** [comment]
@rupeshkoushik07  requesting your review, thanks!

**Review by pranav-bhatt — APPROVED** [review]

File: src/safeposix/syscalls/net_calls.rs [hunk 12]
```diff
@@ -2298,23 +2439,31 @@ impl Cage {
...
+                    //Check if there are any pending incoming connections to the listening socket
+                    //and grab the incoming connection's socket raw fd and its address
                     let mut pendingvec =
                         NET_METADATA.pending_conn_table.get_mut(&porttuple).unwrap();
                     let pendingoption = pendingvec.pop();
                     let (acceptedresult, remote_addr) = match pendingoption {
                         Some(pendingtup) => pendingtup,
                         None => {
                             //unwrap ok because listening
+                            // ** What does the above comment refer to ?? ** //
```
Comment (davidge20) [inline src/safeposix/syscalls/net_calls.rs]: This ^^

File: src/safeposix/syscalls/net_calls.rs [hunk 13]
```diff
@@ -2351,19 +2504,20 @@ impl Cage {
                                 //the case that this recv timeout was exceeded, and we
                                 //should thus not treat this as a failure in our emulated
                                 //socket; see comment in Socket::new in interface/comm.rs
+
+                                //** What does recv timout have to do with accept?? **/
```
Comment (davidge20) [inline src/safeposix/syscalls/net_calls.rs]: This ^^

File: src/safeposix/syscalls/net_calls.rs (hunk 12 above)
Comment (davidge20) [reply src/safeposix/syscalls/net_calls.rs]: delete comment

File: src/safeposix/syscalls/net_calls.rs (hunk 13 above)
Comment (davidge20) [reply src/safeposix/syscalls/net_calls.rs]: delete ref to recv

**Comment by davidge20** [comment]
@yashaswi2000  requesting your review, thanks!

**Review by yashaswi2000 — COMMENTED** [review]
looks mostly good, we dont have any tests for this?

**Comment by rennergade** [comment]
> looks mostly good, we dont have any tests for this?

Yeah we should test for common errors. I think there are enough tests for cases where this works but worth spending a few minutes brainstorming edge cases.

**Review by yashaswi2000 — APPROVED** [review]
LGTM

**Review by rennergade — APPROVED** [review]
